Feat/oss v2 2 metrics (#3762)
influxdata/docs-v2: content/influxdb/v2.2/reference/internals/metrics.md

--
```

## InfluxDB storage statistics

To learn how InfluxDB writes, stores, and caches data, see [InfluxDB storage engine](/influxdb/v2.2/reference/internals/storage-engine/).

### Bucket measurement number

> Gauge of measurement cardinality per bucket.

Reviewer: Number of measurements in a bucket.

#### Example

```sh
# HELP storage_bucket_measurement_num Gauge of measurement cardinality per bucket
# TYPE storage_bucket_measurement_num gauge
storage_bucket_measurement_num{bucket="0c3dd7d2d97f4b23"} 4
--
```

### Bucket series number

> Gauge of series cardinality per bucket.

Reviewer: ```suggestion
Series cardinality per bucket.
```

#### Example

```sh
# HELP storage_bucket_series_num Gauge of series cardinality per bucket
# TYPE storage_bucket_series_num gauge
storage_bucket_series_num{bucket="0c3dd7d2d97f4b23"} 38
--
```

### Cache disk bytes

> Gauge of the size of the most recent [snapshot](/influxdb/v2.2/reference/internals/storage-engine/#cache).

Reviewer: ```suggestion
Size (in bytes) of the most recent [snapshot](/influxdb/v2.2/reference/internals/storage-engine/#cache).
```

#### Example

```sh
# HELP storage_cache_disk_bytes Gauge of size of most recent snapshot
# TYPE storage_cache_disk_bytes gauge
storage_cache_disk_bytes{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="561",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/561",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/561"} 0
--
```

### Cache in use bytes

> Gauge of current memory consumption of the [cache](/influxdb/v2.2/reference/internals/storage-engine/#cache).

Reviewer: ```suggestion
Current memory consumption (in bytes) of the [cache](/influxdb/v2.2/reference/internals/storage-engine/#cache).
```

#### Example

```sh
# HELP storage_cache_inuse_bytes Gauge of current memory consumption of cache
# TYPE storage_cache_inuse_bytes gauge
storage_cache_inuse_bytes{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="561",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/561",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/561"} 0
--
```

### Cache latest snapshot

[Unix time](/influxdb/v2.2/reference/glossary/#unix-timestamp) of the most recent [snapshot](/influxdb/v2.2/reference/internals/storage-engine/#cache).

#### Example

```sh
# HELP storage_cache_latest_snapshot Unix time of most recent snapshot
# TYPE storage_cache_latest_snapshot gauge
storage_cache_latest_snapshot{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="561",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/561",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/561"} 1.644269658196893e+09
--
```

### Cache writes with dropped points

> Counter of [cached](/influxdb/v2.2/reference/internals/storage-engine/#cache) writes that had [rejected points](/influxdb/v2.2/reference/glossary/#rejected-point). Writes with rejected points also increment the [write errors counter (`storage_cache_writes_err`)](#cache-writes-failed).

Reviewer: ```suggestion
Cumulative number of [cached](/influxdb/v2.2/reference/internals/storage-engine/#cache) writes that had [rejected points](/influxdb/v2.2/reference/glossary/#rejected-point). Writes with rejected points also increment the [write errors counter (`storage_cache_writes_err`)](#cache-writes-failed).
```

#### Example

```sh
# HELP storage_cache_writes_dropped Counter of writes to cache with some dropped points
# TYPE storage_cache_writes_dropped counter
storage_cache_writes_dropped{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="561",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/561",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/561"} 0
--
```

### Cache writes failed

> Counter of [cached](/influxdb/v2.2/reference/internals/storage-engine/#cache) writes that [failed](/influxdb/v2.0/write-data/troubleshoot/#troubleshoot-failures), inclusive of [cache writes with dropped points (`storage_cache_writes_dropped`)](#cache-writes-with-dropped-points).

Reviewer: ```suggestion
Cumulative number of [cached](/influxdb/v2.2/reference/internals/storage-engine/#cache) writes that [failed](/influxdb/v2.0/write-data/troubleshoot/#troubleshoot-failures), inclusive of [cache writes with dropped points (`storage_cache_writes_dropped`)](#cache-writes-with-dropped-points).
```

#### Example

```sh
# HELP storage_cache_writes_err Counter of failed writes to cache
# TYPE storage_cache_writes_err counter
storage_cache_writes_err{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="561",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/561",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/561"} 0
--
```

### Cache writes total

> Counter of total writes to [cache](/influxdb/v2.2/reference/internals/storage-engine/#cache).

Reviewer: ```suggestion
Cumulative number of total writes to [cache](/influxdb/v2.2/reference/internals/storage-engine/#cache).
```

#### Example

```sh
# HELP storage_cache_writes_total Counter of all writes to cache
# TYPE storage_cache_writes_total counter
storage_cache_writes_total{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="561",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/561",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/561"} 0
--
```

### Compactions active

> Gauge of currently running [TSM](/influxdb/v2.2/reference/internals/storage-engine/#time-structured-merge-tree-tsm) compactions (by level).

Reviewer: ```suggestion
Number of currently running [TSM](/influxdb/v2.2/reference/internals/storage-engine/#time-structured-merge-tree-tsm) compactions (by level).
```

storage_compactions_duration_seconds_bucket{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="567",level="cache",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/567",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/567",le="6000"} 1
storage_compactions_duration_seconds_bucket{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="567",level="cache",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/567",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/567",le="+Inf"} 1
storage_compactions_duration_seconds_sum{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="567",level="cache",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/567",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/567"} 0.167250668
storage_compactions_duration_seconds_count{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="567",level="cache",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/567",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/567"} 1
--
```

### Compactions failed

> Counter of failed [TSM](/influxdb/v2.2/reference/internals/storage-engine/#time-structured-merge-tree-tsm) compactions (by level).

Reviewer: ```suggestion
Cumulative number of failed [TSM](/influxdb/v2.2/reference/internals/storage-engine/#time-structured-merge-tree-tsm) compactions (by level).
```

#### Example

```sh
# HELP storage_compactions_failed Counter of TSM compactions (by level) that have failed due to error
# TYPE storage_compactions_failed counter
storage_compactions_failed{bucket="ec3f82d1de90eddf",engine="tsm1",id="565",level="1",path="/Users/me/.influxdbv2/engine/data/ec3f82d1de90eddf/autogen/565",walPath="/Users/me/.influxdbv2/engine/wal/ec3f82d1de90eddf/autogen/565"} 0
--
```

### Compactions queued

> Counter of queued [TSM](/influxdb/v2.2/reference/internals/storage-engine/#time-structured-merge-tree-tsm) compactions (by level).

Reviewer: ```suggestion
Number of queued [TSM](/influxdb/v2.2/reference/internals/storage-engine/#time-structured-merge-tree-tsm) compactions (by level).
```

--
```

### Retention check duration

Histogram of retention policy check duration (in seconds).

#### Example

```sh
# HELP storage_retention_check_duration Histogram of duration of retention check (in seconds)
# TYPE storage_retention_check_duration histogram
storage_retention_check_duration_bucket{le="0.005"} 1
storage_retention_check_duration_bucket{le="0.01"} 1
storage_retention_check_duration_bucket{le="0.025"} 1
storage_retention_check_duration_bucket{le="0.05"} 1
storage_retention_check_duration_bucket{le="0.1"} 1
storage_retention_check_duration_bucket{le="0.25"} 1
storage_retention_check_duration_bucket{le="0.5"} 1
storage_retention_check_duration_bucket{le="1"} 1
storage_retention_check_duration_bucket{le="2.5"} 1
storage_retention_check_duration_bucket{le="5"} 1
storage_retention_check_duration_bucket{le="10"} 1
storage_retention_check_duration_bucket{le="+Inf"} 1
storage_retention_check_duration_sum 0.000351857
storage_retention_check_duration_count 1
--
```

### Shard disk size

> Gauge of disk size (in bytes) for the [shard](/influxdb/v2.2/reference/internals/shards/).

Reviewer: ```suggestion
Disk size (in bytes) for a [shard](/influxdb/v2.2/reference/internals/shards/).
```

#### Example

```sh
# HELP storage_shard_disk_size Gauge of the disk size for the shard
# TYPE storage_shard_disk_size gauge
storage_shard_disk_size{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="561",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/561",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/561"} 4.188743e+06
--
```

### Shard fields created

Number of [shard](/influxdb/v2.2/reference/internals/shards/) fields created.

#### Example

```sh
# HELP storage_shard_fields_created Counter of the number of fields created
# TYPE storage_shard_fields_created counter
storage_shard_fields_created{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="561",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/561",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/561"} 0
--
```

### Shard series

> Gauge of the number of series in the [shard](/influxdb/v2.2/reference/internals/shards/) index.

Reviewer: ```suggestion
Number of series in a [shard](/influxdb/v2.2/reference/internals/shards/) index.
```

#### Example

```sh
# HELP storage_shard_series Gauge of the number of series in the shard index
# TYPE storage_shard_series gauge
storage_shard_series{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="561",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/561",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/561"} 38
--
```

### Shard writes

> Number of [shard write](/influxdb/v2.2/reference/internals/shards/#shard-writes) requests.

Reviewer: ```suggestion
Cumulative number of [shard write](/influxdb/v2.2/reference/internals/shards/#shard-writes) requests.
```

#### Example

```sh
# HELP storage_shard_write_count Count of the number of write requests
# TYPE storage_shard_write_count counter
storage_shard_write_count{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="561",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/561",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/561"} 0
--
```

### Shard dropped points

> Number of [rejected points](/influxdb/v2.2/reference/glossary/#rejected-point) in [shard writes](/influxdb/v2.2/reference/internals/shards/#shard-writes).

Reviewer: ```suggestion
Cumulative number of [rejected points](/influxdb/v2.2/reference/glossary/#rejected-point) in [shard writes](/influxdb/v2.2/reference/internals/shards/#shard-writes).
```

#### Example

```sh
# HELP storage_shard_write_dropped_sum Counter of the number of points dropped
# TYPE storage_shard_write_dropped_sum counter
storage_shard_write_dropped_sum{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="561",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/561",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/561"} 0
--
```

### Shard writes with errors

> Number of [shard write](/influxdb/v2.2/reference/internals/shards/#shard-writes) requests with errors.

Reviewer: ```suggestion
Cumulative number of [shard write](/influxdb/v2.2/reference/internals/shards/#shard-writes) requests with errors.
```

#### Example

```sh
# HELP storage_shard_write_err_count Count of the number of write requests with errors
# TYPE storage_shard_write_err_count counter
storage_shard_write_err_count{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="561",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/561",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/561"} 0
--
```

### Points in shard writes with errors

> Number of points in [shard write](/influxdb/v2.2/reference/internals/shards/#shard-writes) requests with errors.

Reviewer: ```suggestion
Cumulative number of points in [shard write](/influxdb/v2.2/reference/internals/shards/#shard-writes) requests with errors.
```

#### Example

```sh
# HELP storage_shard_write_err_sum Counter of the number of points for write requests with errors
# TYPE storage_shard_write_err_sum counter
storage_shard_write_err_sum{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="561",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/561",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/561"} 0
--
```

### Points in shard writes

> Number of points in [shard write](/influxdb/v2.2/reference/internals/shards/#shard-writes) requests.

Reviewer: ```suggestion
Cumulative number of points in [shard write](/influxdb/v2.2/reference/internals/shards/#shard-writes) requests.
```

#### Example

```sh
# HELP storage_shard_write_sum Counter of the number of points for write requests
# TYPE storage_shard_write_sum counter
storage_shard_write_sum{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="561",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/561",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/561"} 0
--
```

### Shard data size

> Gauge of the data size (in bytes) for each [shard](/influxdb/v2.2/reference/internals/shards/).

Reviewer: ```suggestion
Data size (in bytes) for a [shard](/influxdb/v2.2/reference/internals/shards/).
```

#### Example

```sh
# HELP storage_tsm_files_disk_bytes Gauge of data size in bytes for each shard
# TYPE storage_tsm_files_disk_bytes gauge
storage_tsm_files_disk_bytes{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="561",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/561",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/561"} 4.188743e+06
--
```

### Shard files

> Gauge of the number of files per [shard](/influxdb/v2.2/reference/internals/shards/).

Reviewer: ```suggestion
Number of files in a [shard](/influxdb/v2.2/reference/internals/shards/).
```

#### Example

```sh
# HELP storage_tsm_files_total Gauge of number of files per shard
# TYPE storage_tsm_files_total gauge
storage_tsm_files_total{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="561",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/561",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/561"} 1
--
```

### WAL size

> Gauge of the [WAL](/influxdb/v2.2/reference/internals/storage-engine/#write-ahead-log-wal) size (in bytes).

Reviewer: ```suggestion
[WAL](/influxdb/v2.2/reference/internals/storage-engine/#write-ahead-log-wal) size (in bytes).
```

#### Example

```sh
# HELP storage_wal_size Gauge of size of WAL in bytes
# TYPE storage_wal_size gauge
storage_wal_size{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="561",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/561",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/561"} 0
--
```

## WAL write attempts

> Number of write attempts to the [WAL](/influxdb/v2.2/reference/internals/storage-engine/#write-ahead-log-wal).

Reviewer: ```suggestion
Cumulative number of write attempts to the [WAL](/influxdb/v2.2/reference/internals/storage-engine/#write-ahead-log-wal).
```

#### Example

```sh
# HELP storage_wal_writes Number of write attempts to the WAL
# TYPE storage_wal_writes counter
storage_wal_writes{bucket="0c3dd7d2d97f4b23",engine="tsm1",id="561",path="/Users/me/.influxdbv2/engine/data/0c3dd7d2d97f4b23/autogen/561",walPath="/Users/me/.influxdbv2/engine/wal/0c3dd7d2d97f4b23/autogen/561"} 0
--
```

## WAL failed write attempts

> Number of failed write attempts to the [WAL](/influxdb/v2.2/reference/internals/storage-engine/#write-ahead-log-wal).

Reviewer: ```suggestion
Cumulative number of failed write attempts to the [WAL](/influxdb/v2.2/reference/internals/storage-engine/#write-ahead-log-wal).
```

storage_writer_ok_points_bucket{path="/Users/me/.influxdbv2/engine",le="+Inf"} 24
storage_writer_ok_points_sum{path="/Users/me/.influxdbv2/engine"} 125787
storage_writer_ok_points_count{path="/Users/me/.influxdbv2/engine"} 24
--
```

## Points in write requests

Histogram of the number of points in write requests.

#### Example

```sh
# HELP storage_writer_req_points Histogram of number of points requested to be written
# TYPE storage_writer_req_points histogram
storage_writer_req_points_bucket{path="/Users/me/.influxdbv2/engine",le="10"} 6
storage_writer_req_points_bucket{path="/Users/me/.influxdbv2/engine",le="100"} 6
storage_writer_req_points_bucket{path="/Users/me/.influxdbv2/engine",le="1000"} 6
storage_writer_req_points_bucket{path="/Users/me/.influxdbv2/engine",le="10000"} 14
storage_writer_req_points_bucket{path="/Users/me/.influxdbv2/engine",le="100000"} 18
storage_writer_req_points_bucket{path="/Users/me/.influxdbv2/engine",le="+Inf"} 18
storage_writer_req_points_sum{path="/Users/me/.influxdbv2/engine"} 125787
storage_writer_req_points_count{path="/Users/me/.influxdbv2/engine"} 18
--
```

## Shard write request timeouts

> Number of [shard write](/influxdb/v2.2/reference/internals/shards/#shard-writes) request timeouts.

Reviewer: ```suggestion
Cumulative number of [shard write](/influxdb/v2.2/reference/internals/shards/#shard-writes) request timeouts.
```
